react/src/layout/report.tsx

```diff
@@ -261,12 +261,21 @@ export function ReportAnimationController(props: { animationReader: AnimationRea
 }
 
 export type SimulationStartConfig = {
-    reportGroupName: string
+    reportGroupName: string,
+    items: SchemaLayout[]
 }
 
 export class SimulationStartLayout extends Layout<SimulationStartConfig, {}> {
+    childLayouts: Layout[];
+    
+    constructor(config: SimulationStartConfig) {
+        super(config);
+
+        this.childLayouts = (config.items || []).map(LAYOUTS.getLayoutForSchema);
```

Comment:
can SimulationStartConfig items be given a default value of [] so we don't have to have this `|| []` in react?

Reply:
It can but this would be just as long and would be placed on the line above. Types cannot have default values since they aren't classes and the objects following them aren't derived from the type, they just follow them. I could place `config.items = config.items || []` on line 273 but since this is a mutable type I would also need to do `config = {...config}` to create a copy first on line 272. Easier to work with the data we have instead of mutating it first thing.

Reply:
> Types cannot have default values since they aren't classes 

This is concerning. There's power in defaults including future proofing code. Is there no way around this?

Reply:
>>Types cannot have default values since they aren't classes

>This is concerning. There's power in defaults including future proofing code. Is there no way around this?

Is there a reason to use a type here instead of a class?

Reply:
> Is there a reason to use a type here instead of a class?

Yes, one of the big advantages of duck typing over classes is that we don't need to keep a dictionary of classes we should be using while reading in data. We could do it either way but I tend to avoid complicated approaches like this for things like defaults. Typescript also lets us mark which of the values could possibly be null with a ?.

Reply:
> one of the big advantages of duck typing over classes is that we don't need to keep a dictionary of classes we should be using while reading in data

Can you elaborate more on that? Why/where would we need to keep a dictionary of classes? 

I don't feel like this is really duck typing either. If config could quack it would have items initialized.

> We could do it either way but I tend to avoid complicated approaches like this for things like defaults

My concern with this bit of code is that it is a pattern we have in our angularjs app that, while simple when looking at one case, becomes complex when we look at all cases. Every caller is going to have to do this check. By distributing that confusion around the code things get more complex. Obviously, in this case it only happens here. But, people copy these patterns and code tends to get used more and more as time goes on. If we solve the problem once for everything then all aspects of the code benefit.